cmd/k8s-operator/ingress.go

```diff
@@ -109,6 +117,10 @@ func (a *IngressReconciler) maybeCleanup(ctx context.Context, logger *zap.Sugare
 // This function adds a finalizer to ing, ensuring that we can handle orderly
 // deprovisioning later.
 func (a *IngressReconciler) maybeProvision(ctx context.Context, logger *zap.SugaredLogger, ing *networkingv1.Ingress) error {
+	if err := a.validateIngressClass(ctx); err != nil {
+		logger.Warnf("error validating tailscale IngressClass: %v. In future this might be a terminal error.", err)
```

Comment:
This appears at default ('info') logs
```
{"level":"warn","ts":"2024-01-11T12:11:18Z","logger":"ingress-reconciler","msg":"error validating tailscale IngressClass: ingressclass.kubernetes.io/is-default-class annotation is set on 'tailscale' IngressClass, but Tailscale Ingress controller does not support default Ingress class. Ensure that you are using 'tailscale' IngressClass from latest Tailscale installation manifests.. In future this might be a terminal error.","ingress-ns":"default","ingress-name":"kuardfoobar"}
```